### PR TITLE
fix: correct encoding with UTF8

### PR DESCRIPTION
**Pull request by moreal**

### CODE REVIEW

File: RocksDbSharp/RocksDbException.cs [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 using System.Runtime.InteropServices;
+using System.Text;
 
 namespace RocksDbSharp
 {
     public class RocksDbException : RocksDbSharpException
     {
         public RocksDbException(IntPtr errptr)
-            : base(Marshal.PtrToStringUni(errptr))
+            : base(Encoding.UTF8.GetString(Encoding.Unicode.GetBytes(Marshal.PtrToStringUni(errptr))))
```
Comment (dahlia) [inline RocksDbSharp/RocksDbException.cs]: How about this?

```suggestion
            : base(Marshal.PtrToStringUTF8(errptr))
```

File: RocksDbSharp/RocksDbException.cs [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
         {
+            Marshal.PtrToStringUni(errptr);
```
Comment (dahlia) [inline RocksDbSharp/RocksDbException.cs]: Why is this necessary?

Comment (moreal) [reply RocksDbSharp/RocksDbException.cs]: No, it is my mistake during debugging. I will remove it 😥

File: RocksDbSharp/RocksDbException.cs (hunk 1 above)
Comment (moreal) [reply RocksDbSharp/RocksDbException.cs]: It seems not available at the current target framework, `netstandard2.0`. So it needs to bump it into `netstandard2.1`. Is there no problem when using Unity? 🤔

Comment (dahlia) [reply RocksDbSharp/RocksDbException.cs]: Okay, then we should find the alternative way.  To me, your first attempt seems counterintuitive, because the intention of *reading a UTF-16-encoded bytes (`StringUni`) into `string` from an `IntPtr`, and then turning it to UTF-16-encoded bytes again* is difficult to find.  What we actually want to do here is to read null-terminating string into `byte[]`, but there seems no such method in `Marshal` class, so we do round-trip as a workaround instead to get a raw `byte[]` of the error message…

Personally I believe it's better to have its own method to read a null-terminating `byte[]` from `IntPtr` instead, and use it instead of such round-trip between bytes and Unicode string, what's your thoughts?

----

Here's my untested pseudo code:

```csharp
static unsafe byte[] ReadNullTerminatingBytesFromIntPtr(IntPtr ptr)
{
    int size = 0;
    for (byte* p = ptr; *p != 0; p++)
    {
        size++;
    }

    var buffer = new byte[size];
    return Marshal.Copy(ptr, buffer, 0, size);
}
```

Comment (moreal) [reply RocksDbSharp/RocksDbException.cs]: I agree, the way you said seems better 😀

**Review by dahlia — APPROVED** [review]

**Review by longfin — APPROVED** [review]

**Review by riemannulus — APPROVED** [review]
